pennylane/templates/subroutines/hilbert_schmidt.py

```diff
@@ -284,8 +289,10 @@ def compute_decomposition(
             decomp_ops.extend(u_tape.operations)
 
         # Unitary V conjugate
-        decomp_ops.extend(qml.adjoint(qml.apply, lazy=False)(op_v) for op_v in v_tape.operations)
-
+        decomp_ops.extend(
+            qml.adjoint(qml.apply, lazy=False)(op_v) for op_v in reversed(v_tape.operations)
+        )
+        # Single qubit measurement
```

Comment:
I think so because the only difference between the Local and non-local Hilbert-Schmidt test is that, in the latter, only two of the *2n* qubits are measured at the end. 

Here we are reversing the order of the operations in *V*, which should be done in both because the *V* quantum function is treated the same way in both tests.

Reply:
> Is this correct?

I'm slightly confused about which part could be wrong here?